Platform/TigerlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -2415,9 +2416,31 @@ UpdateCpuNvs (
     CpuNvs->DtsIoTrapLength = DTS_IO_TRAP_LENGTH;
     CpuNvs->DtsAcpiEnable = DTS_ACPI_DISABLE;
   }
+
+  ///
+  /// Update TXT status for ACPI
+  ///
+  CpuNvs->TxtEnabled = 0;  // Default to disabled
+  if (FeaturePcdGet (PcdTxtEnabled)) {
+    FEATURES_CFG_DATA *FeaturesCfgData;
+    FeaturesCfgData = (FEATURES_CFG_DATA *) FindConfigDataByTag(CDATA_FEATURES_TAG);
+    if ((FeaturesCfgData != NULL) && (FeaturesCfgData->Features.TxtEnabled == 1)) {
+      UINT64 Ia32FeatureControl;
+      // Read MSR 0x3A (IA32_FEATURE_CONTROL) to check if TXT is actually enabled
+      Ia32FeatureControl = AsmReadMsr64 (MSR_IA32_FEATURE_CONTROL);
+      // Check if VMX and SMX are enabled (TXT_OPT_IN_VMX_AND_SMX_MSR_VALUE = 0xFF03)
+      if ((Ia32FeatureControl & 0xFF03) == 0xFF03) {
```

Comment:
In the TxtLib, there is an internal function IsTxtEnabled(),  do you think it is better to expose it for use here? This API could be updated only checking MSR as you did here, without asking any parameters.

Reply:
Guo, can we have all CPU NVS updates at oneplace?